src/include/bootloader/routines/misc.rb

```diff
@@ -139,35 +139,51 @@ def GetBootloaderDevices
       ret
     end
 
-    # get kernel parameter from kernel command line
+    # get kernel parameter values from kernel command line
+    #
     # @param [String] line string original line
     # @param [String] key string parameter key
-    # @return [String] value, "false" if not present,
-    #   "true" if present key without value
+    # @return [String,Array<String>] value, "false" if not present or "true" if present without
+    #   value. If the parameter has on more than 1 value, an array will be returned.
     def getKernelParamFromLine(line, key)
       # FIXME: this doesn't work with quotes and spaces
       res = "false"
       # we can get nil if params is not yet proposed, so return not there (bnc#902397)
       return res unless line
       params = line.split(" ").reject(&:empty?)
-      params.each do |p|
-        l = p.split("=")
-        res = l[1] || "true" if l[0] == key
+      values = params.map { |p| kernel_param_parts(p) }.select { |p| p[0] == key }.map(&:last)
```

Comment:
hmm, this is quite confusing, as it is quite hidden that kernel_param_parts remap to "true" if value is not there. So prefer more old code, where it is more obvious. Change also have one nasty side effect. For flags like "verbose" it do not make sense to report it multipletime so for line like `verbose crashkernel=bla,bla verbose 3` it should return only "verbose" => "true" not ["true", "true"] as it do not make sense.